src/azul/chalice.py

```diff
@@ -504,34 +497,105 @@ def swagger_resource(self, file) -> Response:
+        @property
+        def alarm_tf_resource_name(self) -> str:
+            return self.lambda_tf_resource_name + '_' + self.metric.name
+
+    @attr.s(auto_attribs=True, frozen=True, kw_only=True)
+    class retry:  # noqa: N801
+        """
+        Use this decorator to specify the number of times a Lambda invocation of
+        the decorated event handler function should be retried. This decorator
+        cannot be used to decorate view functions, i.e. functions also decorated
+        with ``@app.route``.
+
+        https://docs.aws.amazon.com/lambda/latest/dg/invocation-retries.html
+        """
+        value: int
+        handler_name: str | None = attr.ib(default=None)
+
+        def __call__(self, f):
+            assert isinstance(f, chalice.app.EventSourceHandler), f
+            f.retries = self.value
+            return f
 
     @property
-    def metric_thresholds(self) -> list[MetricThreshold]:
-        default_threshold = 0
-        thresholds = []
+    def metric_alarms(self) -> list[metric_alarm]:
+        alarms = []
         lambda_name, _ = config.unqualified_resource_name(self.app_name)
+
         for metric in LambdaMetric:
-            # The api_handler lambda functions (indexer & service) aren't included
-            # in the app_module's handler_map, so we account for those first.
-            thresholds.append(MetricThreshold(lambda_name=lambda_name,
-                                              metric=metric,
-                                              value=default_threshold))
-            for handler_name, handler in self.handler_map.items():
-                if isinstance(handler, chalice.app.EventSourceHandler):
-                    threshold = getattr(handler, f'{metric.name}_threshold', default_threshold)
-                    thresholds.append(MetricThreshold(lambda_name=lambda_name,
-                                                      handler_name=handler_name,
-                                                      metric=metric,
-                                                      value=threshold))
-        return thresholds
+            # The api_handler lambda functions (indexer & service) aren't
+            # included in the app_module's handler_map, so we account for those
+            # first.
+            alarms.append(self.metric_alarm(metric=metric).bind(lambda_name))
+        for handler_name, handler in self.handler_map.items():
+            if isinstance(handler, chalice.app.EventSourceHandler):
+                try:
+                    metric_alarms = getattr(handler, 'metric_alarms')
+                except AttributeError:
+                    alarms.extend(
+                        self.metric_alarm(metric=metric).bind(lambda_name, handler_name)
+                        for metric in LambdaMetric
+                    )
+                else:
+                    for metric_alarm in metric_alarms:
+                        alarms.append(metric_alarm.bind(lambda_name, handler_name))
```

Comment:
I believe this can be simplified to 

```suggestion
                try:
                    metric_alarms = getattr(handler, 'metric_alarms')
                except AttributeError:
                    metric_alarms = (self.metric_alarm(metric=metric) for metric in LambdaMetric) 
                for metric_alarm in metric_alarms:
                    alarms.append(metric_alarm.bind(lambda_name, handler_name))
```

Reply:
... or even

```suggestion
                for metric_alarm in getattr(handler, 
                                            'metric_alarms', 
                                            (self.metric_alarm(metric=metric) for metric in LambdaMetric)):
                    alarms.append(metric_alarm.bind(lamba_name, handler_name))
```

but that might be too compact to read clearly.